metals/src/main/scala/scala/meta/internal/metals/BuildServerConnection.scala

```diff
@@ -265,9 +266,11 @@ object BuildServerConnection {
 
     def setupServer(): Future[LauncherConnection] = {
       connect().map { case conn @ SocketConnection(_, output, input, _, _) =>
-        val tracePrinter = GlobalTrace.setupTracePrinter("BSP")
+        val tracePrinter =
+          if (workspace.isMetaWorkspace) None
+          else Trace.setupTracePrinter("BSP")
```

Comment:
Could we instead of adding the new `Workspace` entity, add a second entry here that would be the path. We could then write to `.metals` directory in sbt project if needed. This way we will not override the default BSP connection trace and we can also debug the other connections if we want.